MekHQ month advancement issue

### Environment
MekHQ unstable 0.49.2 also encountered on 0.49.1
Windows 10 home 64-bit
Java version 16.0.1

### Description
When the advance day button is clicked to advance a month during a contract, the game fails to advance to the 1st of the next month and only adds entries to the bottom of the daily activity log for the various markets updating. No other new month activity seems to occur (i.e. no money paid out, no exp, no events). Date will stay on the say before the end of the month, then after the advance day button is clicked a second time, it will change to the second day of the month.
![Example date failure](https://user-images.githubusercontent.com/86205253/122686147-c4859b00-d1dd-11eb-99e9-2523aae07fde.png)

Can be easily replicated by just advancing the day, no need to interact with the scheduled scenario. Whether it is completed or not the bug occurs.

### Files

[megameklog.txt](https://github.com/MegaMek/mekhq/files/6682981/megameklog.txt)
[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/6682982/mekhqlog.txt)
[Garett's Fusiliers in progress.zip](https://github.com/MegaMek/mekhq/files/6682996/Garett.s.Fusiliers.in.progress.zip)

Two exceptions of note:
```
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at megamek.common.event.EventListener.trigger(EventListener.java:33)
	at megamek.common.event.EventBus.internalTrigger(EventBus.java:158)
	at megamek.common.event.EventBus.trigger(EventBus.java:147)
	at mekhq.MekHQ.triggerEvent(MekHQ.java:680)
	at mekhq.campaign.Campaign.newDay(Campaign.java:3406)
	at mekhq.campaign.CampaignController.advanceDay(CampaignController.java:82)
	at mekhq.gui.CampaignGUI.lambda$initTopButtons$42(CampaignGUI.java:1180)
...
Caused by: java.lang.NullPointerException: Cannot invoke "java.time.LocalDate.isEqual(java.time.chrono.ChronoLocalDate)" because the return value of "mekhq.campaign.mission.AtBScenario.getDate()" is null
	at mekhq.gui.dialog.OutstandingScenariosNagDialog.lambda$checkNag$0(OutstandingScenariosNagDialog.java:44)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1602)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:632)
	at mekhq.gui.dialog.OutstandingScenariosNagDialog.lambda$checkNag$1(OutstandingScenariosNagDialog.java:44)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1602)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:632)
	at mekhq.gui.dialog.OutstandingScenariosNagDialog.checkNag(OutstandingScenariosNagDialog.java:43)
	at mekhq.gui.baseComponents.AbstractMHQNagDialog.<init>(AbstractMHQNagDialog.java:43)
	at mekhq.gui.dialog.OutstandingScenariosNagDialog.<init>(OutstandingScenariosNagDialog.java:31)
	at mekhq.gui.CampaignGUI.handleDayEnding(CampaignGUI.java:2612)
	... 47 more
```
Second exception:
```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException: Cannot invoke "java.time.chrono.ChronoLocalDate.toEpochDay()" because "other" is null
	at java.base/java.time.chrono.ChronoLocalDate.isAfter(ChronoLocalDate.java:729)
	at java.base/java.time.LocalDate.isAfter(LocalDate.java:2059)
	at mekhq.campaign.mission.AtBContract.checkMorale(AtBContract.java:305)
	at mekhq.campaign.Campaign.processNewDayATB(Campaign.java:3179)
	at mekhq.campaign.Campaign.newDay(Campaign.java:3446)
	at mekhq.campaign.CampaignController.advanceDay(CampaignController.java:82)
	at mekhq.gui.CampaignGUI.lambda$initTopButtons$42(CampaignGUI.java:1180)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972)
...
```

--- a/MekHQ/src/mekhq/campaign/mission/AtBContract.java
+++ b/MekHQ/src/mekhq/campaign/mission/AtBContract.java
@@ -302,7 +302,7 @@ public class AtBContract extends Contract implements Serializable {
         LocalDate lastMonth = today.minusMonths(1);
 
         for (Scenario s : getScenarios()) {
-            if (lastMonth.isAfter(s.getDate())) {
+            if ((s.getDate() != null) && lastMonth.isAfter(s.getDate())) {
                 continue;
             }
 

--- a/MekHQ/src/mekhq/campaign/mission/Scenario.java
+++ b/MekHQ/src/mekhq/campaign/mission/Scenario.java
@@ -31,6 +31,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 
+import megamek.common.annotations.Nullable;
 import mekhq.campaign.mission.enums.ScenarioStatus;
 import org.w3c.dom.NamedNodeMap;
 import org.w3c.dom.Node;
@@ -127,12 +128,12 @@ public class Scenario implements Serializable {
         this.status = status;
     }
 
-    public void setDate(LocalDate d) {
-        this.date = d;
+    public @Nullable LocalDate getDate() {
+        return date;
     }
 
-    public LocalDate getDate() {
-        return date;
+    public void setDate(final @Nullable LocalDate date) {
+        this.date = date;
     }
 
     public boolean hasObjectives() {
@@ -341,7 +342,7 @@ public class Scenario implements Serializable {
         if (null != date) {
             MekHqXmlUtil.writeSimpleXmlTag(pw1, indent + 1, "date", MekHqXmlUtil.saveFormattedDate(date));
         }
-        
+
         MekHqXmlUtil.writeSimpleXmlTag(pw1, indent + 1, "cloaked", isCloaked());
     }
 

--- a/MekHQ/src/mekhq/gui/dialog/OutstandingScenariosNagDialog.java
+++ b/MekHQ/src/mekhq/gui/dialog/OutstandingScenariosNagDialog.java
@@ -41,6 +41,7 @@ public class OutstandingScenariosNagDialog extends AbstractMHQNagDialog {
         return !MekHQ.getMekHQOptions().getNagDialogIgnore(getKey())
                 && campaign.getActiveAtBContracts(true).stream()
                         .anyMatch(contract -> contract.getCurrentAtBScenarios().stream()
-                                .anyMatch(scenario -> scenario.getDate().isEqual(campaign.getLocalDate())));
+                                .anyMatch(scenario -> (scenario.getDate() != null)
+                                        && scenario.getDate().isEqual(campaign.getLocalDate())));
     }
 }

